t/unittest/UnifiedApi/Controller/Locationlogs.t

```diff
@@ -73,16 +73,214 @@ $t->get_ok('/api/v1/locationlogs' => json => { })
+            value => '00:01:02:03:04:06'
+        },
+    },
+    \%values,
+    "Got back 00:01:02:03:04:05"
+);
+
+simple_single_query(
+    {
+        query => {
+            op    => 'starts_with',
+            field => 'mac',
+            value => '00:01',
+        },
+    },
+    \%values,
+    "Got back 00:01:02:03:04:05"
+);
+
+simple_single_query(
+    {
+        query => {
+            op    => 'ends_with',
+            field => 'mac',
+            value => '04:05',
+        },
+    },
+    \%values,
+    "Got back 00:01:02:03:04:05"
+);
+
+simple_single_query(
+    {
+        fields => [qw(mac)],
+        query => {
+            op    => 'equals',
+            field => 'mac',
+            value => '00:01:02:03:04:05'
+        },
+    },
+    { mac => "00:01:02:03:04:05" },
+    "Got back 00:01:02:03:04:05"
+);
+
+sub simple_single_query {
+    my ($query, $value, $msg) = @_;
+    $t->post_ok(
+        '/api/v1/locationlogs/search' => json => $query
+      )
+      ->status_is(200)
+    ;
+
+    my $items = $t->tx->res->json->{items};
+    my $item = $items->[0];
+    delete $item->{id};
+    is_deeply($item, $value, $msg)
+}
 
 #truncate the locationlog table
 pf::dal::locationlog->remove_items();
-  
-  
+
+for my $o (0 .. 255) {
+    my $status = pf::dal::locationlog->create(
+        {
+            %values,
+            mac        => sprintf("01:02:03:04:05:%02x", $o),
+            switch_mac => sprintf('06:07:08:09:0a:%02x', $o),
+            port       => $o + int(rand(85)),
+            vlan       => $o,
+        }
+    );
+}
+
+simple_single_query(
+    {
+        'fields' => [qw(mac)],
+        'limit' => 1,
+        'sort' => ['mac DESC'],
+    },
+    {mac => '01:02:03:04:05:ff'},
+    "Got back 00:01:02:03:04:ff"
+);
+
+$t->post_ok(
+    '/api/v1/locationlogs/search?cursor=1' => json => {
+        'fields' => [qw(mac)],
+        'limit' => 1,
+        'cursor' => 1,
+        'sort' => ['mac'],
+    }
+  )
+  ->status_is(200)
+;
+
+is_deeply($t->tx->res->json->{items}[0], { mac => '01:02:03:04:05:01'});
+
+$t->post_ok(
+    '/api/v1/locationlogs/search' => json => { }
+  )
+  ->status_is(200)
+;
+$items = $t->tx->res->json->{items};
+
+is(@$items, 25, "Returns the defaults items count");
+
+$t->post_ok(
+    '/api/v1/locationlogs/search' => json => { limit => 100 }
+  )
+  ->status_is(200)
+  ->json_is('/nextCursor', 100)
+  ->json_is('/prevCursor', 0)
+;
+$items = $t->tx->res->json->{items};
+
+is(@$items, 100, "Returns 100 items");
+
+$t->post_ok(
+    '/api/v1/locationlogs/search' => json => { limit => 256 }
+  )
+  ->status_is(200)
+  ->json_hasnt('/nextCursor')
+  ->json_is('/prevCursor', 0)
+;
+$items = $t->tx->res->json->{items};
+
+is(@$items, 256, "Returns 256 items");
+
+$t->post_ok(
+    '/api/v1/locationlogs/search' => json => { limit => 267 }
+  )
+  ->status_is(200)
+  ->json_hasnt('/nextCursor')
+  ->json_is('/prevCursor', 0)
+;
+$items = $t->tx->res->json->{items};
+
+is(@$items, 256, "Returns 256 items");
+
```

Comment:
Maybe I'm blind but we're missing a test for using the cursor parameter for pagination

Reply:
https://github.com/inverse-inc/packetfence/pull/2992/files/4b0d2a630f5e269803c225d5dc740ca63d359413#diff-acbd078bd53801f2f2fc209e5c36a09fR230